tests/test_analyze__main.py

```diff
@@ -0,0 +1,112 @@
+# tests/test_analyze__main.py
+#
+# These are end-to-end tests that run module `birdnet_analyzer.analyze`
+# as if calling it from the command-line.
+#
+# When run with certain arguments on known example input,
+# an output file should be generated with expected specific format and data
+# as found in tests/resources/SNAPSHOT.analyze__main.*)
+#
+# How to run these tests;
+# 1. (Prerequisite) from root of this repository run: `pip install -r requirements.txt`
+# 2. From root of this repository run: `pytest tests/test_analyze__main.py`
+
+import subprocess
+import os
+
+def test_analyze_case1_example_min_conf(tmp_path):
+    """
+    Tests a command similar to:
+    python3 analyze.py --i example/ --o example/ --slist example/ --min_conf 0.5 --threads 4
+    """
+    # GIVEN: analyze is called on example/ dir with 1 sound file, soundscape.wav, with a minimum confidence level of 0.5
+    output_dir = tmp_path
+    # comment the following line in in order to write a file to troubleshoot or take a new snapshot
+    # output_dir = 'birdnet_analyzer/example'
```

Comment:
This comment is added, because this code intentionally uses `pytest`’s `tmp_path` mechanism, which cleans up after itself on each test so as not to clutter the developer’s environment or leave behind anything she might mistake for real (non-test) output.